Data/Sys/Shaders/default_pre_post_process.glsl

```diff
@@ -195,214 +255,169 @@ float4 SharpBilinearSample(float3 uvw, float gamma)
+        color = BicubicSample(uvw, CUBIC_COEFF_GEN(0.0, 0.0));
+
+    return color;
 }
 
 void main()
 {
-	// This tries to fall back on GPU HW sampling if it can (it won't be gamma corrected).
-	bool raw_resampling = resampling_method <= 0;
-	bool needs_rescaling = GetResolution() != GetWindowResolution();
-
-	bool needs_resampling = needs_rescaling && (OptionEnabled(hdr_output) || OptionEnabled(correct_gamma) || !raw_resampling);
-
-	float4 color;
-
-	if (needs_resampling)
-	{
-		// Doing linear sampling in "gamma space" on linear texture formats isn't correct.
-		// If the source and target resolutions don't match, the GPU will return a color
-		// that is the average of 4 gamma space colors, but gamma space colors can't be blended together,
-		// gamma neeeds to be de-applied first. This makes a big difference if colors change
-		// drastically between two pixels.
-
-		color = LinearGammaCorrectedSample(game_gamma);
-	}
-	else
-	{
-		// Default GPU HW sampling. Bilinear is identical to Nearest Neighbor if the input and output resolutions match.
-		if (needs_rescaling)
-			color = texture(samp0, v_tex0);
-		else
-			color = texture(samp1, v_tex0);
-
-		// Convert to linear before doing any other of follow up operations.
-		color.rgb = pow(color.rgb, float3(game_gamma));
-	}
-
-	if (OptionEnabled(correct_color_space))
-	{
-		if (game_color_space == 0)
-			color.rgb = color.rgb * from_NTSCM;
-		else if (game_color_space == 1)
-			color.rgb = color.rgb * from_NTSCJ;
-		else if (game_color_space == 2)
-			color.rgb = color.rgb * from_PAL;
-	}
-
-	if (OptionEnabled(hdr_output))
-	{
-		float hdr_paper_white = hdr_paper_white_nits / hdr_sdr_white_nits;
-		color.rgb *= hdr_paper_white;
-	}
-
-	if (OptionEnabled(linear_space_output))
-	{
-		// Nothing to do here
-	}
-	// Correct the SDR gamma for sRGB (PC/Monitor) or ~2.2 (Common TV gamma)
-	else if (OptionEnabled(correct_gamma))
-	{
-		if (OptionEnabled(sdr_display_gamma_sRGB))
-			color.rgb = LinearTosRGBGamma(color.rgb);
-		else
-			color.rgb = pow(color.rgb, float3(1.0 / sdr_display_custom_gamma));
-	}
-	// Restore the original gamma without changes
-	else
-	{
-			color.rgb = pow(color.rgb, float3(1.0 / game_gamma));
-	}
-
-	SetOutput(color);
+    float4 color = float4(0.0, 0.0, 0.0, 1.0);
+
+    if (GetResolution() == GetWindowResolution() || resampling_method == 0)
+        color = sRGBToLinear(texture(samp1, v_tex0));
+    else
+        color = GetResampledColor();
+
+    if (OptionEnabled(correct_color_space)) {
+        if (game_color_space == 0)
+            color.rgb = color.rgb * from_NTSCM;
+        else if (game_color_space == 1)
+            color.rgb = color.rgb * from_NTSCJ;
+        else if (game_color_space == 2)
+            color.rgb = color.rgb * from_PAL;
+    }
+
+    if (OptionEnabled(hdr_output)) {
+        float hdr_paper_white = hdr_paper_white_nits / hdr_sdr_white_nits;
+        color = OklabToLinearRGB(hdr_paper_white * LinearRGBToOklab(color));
```

Comment:
This is wrong: to scale up the brightness of Oklab colors, you should just scale its 3rd component (intensity, which is comparable to brightness/luminance).
Though even after that, it's a very unconventional and unnecessary way of scaling up the brightness from SDR to HDR, and the output brightness wouldn't match the target nits the user had selected (it wouldn't match at all, as Oklab intensity is not in linear space, so multiplying it yields very different results).
Usually brightness is scaled "linearly", with linear space colors, and that's what the code was doing before. It's akin to changing the brightness of your monitor/TV in its settings, or to changing the Windows SDR to HDR brightness slider.
Generally speaking, there's no need to scale colors with Oklab to maintain the original perceived colors (the color our eyes would have perceived at the original SDR brightness, which anyway, wouldn't necessarily be 80 or 100 nits, but it would depend on the TVs the game was designed on).
This is likely to not make much difference because color perception doesn't change much between ~100 and (e.g.) 300 nits, but it's just unnecessary (IMO) and "slower". Unless you can come up with an explanation for doing this, let's avoid it. Even then, it should be a separate (HDR) post process, like my AutoHDR post process shader is.

Reply:
Scaling the 3rd component looks weird and desaturated. Scaling it like this increases perceived exposure.